misc/pytorch_toolkit/breast_ultrasound_simulation/src/solve.py

```diff
@@ -0,0 +1,358 @@
+import torch
+import torch.nn as nn
+import torch.optim as optim
+from torch.utils.tensorboard import SummaryWriter
+import torchvision
+import os
+import numpy as np
+import time
+from model import GeneratorModel, DiscriminatorModel, GeneratorInter, Generator3dInter
+
+device = torch.device('cuda:0' if torch.cuda.is_available() else 'cpu')
```

Comment:
It would be better to pass device as one of the arguments rather then hardcode it as a global variable.